x-pack/plugins/infra/server/routes/snapshot/lib/apply_metadata_to_last_path.ts

```diff
@@ -47,12 +47,12 @@ export const applyMetadataToLastPath = (
         source.configuration.fields
       );
       // Set the label as the name and fallback to the id OR path.value
-      lastPath.label = get(firstMetaDoc, inventoryFields.name, lastPath.value);
+      lastPath.label = (firstMetaDoc[inventoryFields.name] as string | undefined) ?? lastPath.value;
       // If the inventory fields contain an ip address, we need to try and set that
       // on the path object. IP addersses are typically stored as multiple fields. We will
       // use the first IPV4 address we find.
       if (inventoryFields.ip) {
-        const ipAddresses = get(firstMetaDoc, inventoryFields.ip) as string[];
+        const ipAddresses = firstMetaDoc[inventoryFields.ip] as string | undefined;
         if (Array.isArray(ipAddresses)) {
```

Comment:
I think we can remove this array condition if ip is always a string.  CC @simianhacker

Reply:
Based on the comment, I think it can be also an array...

Reply:
I think we need to find out for sure. If so, the type is incorrect to say it must be a string or undefined?

Reply:
I changed it back to how it was (so this change is outdated) and I keep the type as `string[]`.

Reply:
I think we should keep it as it was since we didn't change anything about the types of the fields, so if with t`top_hits` it could return an array I guess with `top_metrics` it can return an array too.
Anyway, I'll try to replicate this use case and verify it.

Reply:
IP and mac addresses are returned as arrays. See note above.